agent/connect/uri_service.go

```diff
@@ -52,3 +53,33 @@ func (id SpiffeIDService) uriPath() string {
 	}
 	return path
 }
+
+// SpiffeIDIdentity is the structure to represent the SPIFFE ID for an identity.
+type SpiffeIDIdentity struct {
+	Host      string
+	Partition string
+	Namespace string
+	Identity  string
+}
+
+func (id SpiffeIDIdentity) URI() *url.URL {
+	var result url.URL
+	result.Scheme = "spiffe"
+	result.Host = id.Host
+	result.Path = fmt.Sprintf("/ap/%s/ns/%s/identity/%s",
+		id.Partition,
+		id.Namespace,
+		id.Identity,
+	)
+	return &result
+}
+
+// SpiffeIDFromIdentityRef creates the SPIFFE ID from an identity.
+func SpiffeIDFromIdentityRef(trustDomain string, ref *pbresource.Reference) string {
```

Comment:
Should you document that only workload identity types are supported and do an EqualType check and panic if the user coded it wrong?

Reply:
Yep but I will need to add it as a TODO because this type doesn't yet exist.